src/vsr/journal.zig

```diff
@@ -1296,10 +1296,9 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             assert(journal.headers.len == cases.len);
 
             // Refine cases @B and @C: Repair (truncate) a prepare if it was torn during a crash.
-            if (journal.recover_torn_prepare(&cases)) |torn_slot| {
+            for (journal.torn_prepares(&cases).const_slice()) |torn_slot| {
```

Comment:
I genuinely don't know whether this code is safe or not, so just to be careful let's extract the array:
```suggestion
            const torn_slots = journal.torn_prepares(&cases);
            for (torn_slots.const_slice()) |torn_slot| {
```

Reply:
Took care of this in https://github.com/tigerbeetle/tigerbeetle/pull/2345/commits/4a8eb61f8001cc24f89b665c5af17b7011e9eb74.